File: 0242-valid-anagram/0242-valid-anagram.java
class Solution {
    public boolean isAnagram(String s, String t) {
        HashMap<Character, Integer> maps = new HashMap<>();
        HashMap<Character, Integer> mapt = new HashMap<>();

        if(s.length() != t.length()) {
            return false;
        }

        for(int i = 0; i < s.length(); i++) {
            char ch = s.charAt(i);

            maps.merge(ch, 1, Integer::sum);
        }

        for(int i = 0; i < t.length(); i++) {
            char ch = t.charAt(i);

            mapt.merge(ch, 1, Integer::sum);
        }

        for(int i = 0; i < s.length(); i++) {
            char ch = s.charAt(i);

            if(!maps.get(ch).equals(mapt.get(ch))) {
                return false;
            }
        } 


        return true; 
    }

}

File: 0242-valid-anagram/0242-valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        smap = {}
        tmap = {}

        if len(s) != len(t):
            return False

        for i in s:
            if i in smap:
                smap[i] += 1
            else:
                smap[i] = 1
        
        for i in t:
            if i in tmap:
                tmap[i] += 1
            else:
                tmap[i] = 1
        print(smap)
        for i in smap:
            if i not in tmap or smap[i] != tmap[i]:
                return False
        return True